nntrainer/src/neuralnet.cpp

```diff
@@ -438,92 +433,54 @@ int NeuralNetwork::setTrainConfig(std::vector<std::string> values) {
 
 int NeuralNetwork::init() {
   int status = ML_ERROR_NONE;
-  bool last = false;
   TensorDim previous_dim, def_init_dim;
 
+  status = checkValidation();
+  NN_RETURN_STATUS();
+
   /** Note: number of entries in layers will change. */
   for (unsigned int i = 0; i < layers.size(); ++i) {
-    if (i == layers.size() - 1)
-      last = true;
-
-    VERIFY_SET_DIMENSION();
-
-    switch (layers[i]->getType()) {
-    case LAYER_IN: {
-      status = layers[i]->initialize(last);
-      NN_RETURN_STATUS();
-    } break;
-    case LAYER_CONV2D: {
-      std::shared_ptr<Conv2DLayer> conv2d_layer =
-        std::static_pointer_cast<Conv2DLayer>(layers[i]);
-
-      status = layers[i]->setCost(cost);
-      NN_RETURN_STATUS();
-
-      status = layers[i]->initialize(last);
-      NN_RETURN_STATUS();
-
-      status = conv2d_layer->setOptimizer(opt);
-      NN_RETURN_STATUS();
```

Comment:
so, we cannot set different optimizer for each layer then. Am I right?

Reply:
This PR does not change the behavior as per setting optimizer. As per now, yes.

Reply:
I think we have plan to try change opt for a layer sooner or later. If we are doing this soon, then we have to reverse current implementation?

Reply:
We do not need to revert this patch. It will need less amount of change after this patch since we only have to change `layer::setOptimizer.` (currently, `conv2d::setOptimizer`, `layer::setOptimizer`, `fc::setOptimizer`...  and so on)